src/github/github.service.ts

```diff
@@ -0,0 +1,106 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { CommunityStats, GithubProfile } from './interfaces/github.interface';
+import { GithubDTO } from './dto/github.dto';
+import { CommunitystatsMappingService } from './communitystats-mapping.service';
+
+@Injectable()
+export class GithubService {
+  constructor(private readonly mappingService: CommunitystatsMappingService) {}
+  private githubProfiles: Array<GithubProfile> = [];
+
+  createGithub(body: GithubDTO): GithubProfile {
+    const newGithubProfile: GithubProfile = {
+      id: 123,
+      username: body.username,
+      avatarUrl: body.avatarUrl,
+      bio: body.bio,
+      communityStats: this.mappingService.mapCommunityState(
+        body.event,
+        {} as CommunityStats,
+      ),
+      followers: body.followers,
+      repos: body.repos,
+      createdOn: new Date('2021-01-01T00:00:00.000Z'),
+      updatedOn: new Date('2021-01-01T00:00:00.000Z'),
+    };
+
+    if (!newGithubProfile.username) {
+      throw new HttpException('Incomplete Data', HttpStatus.BAD_REQUEST);
+    }
+    this.githubProfiles.push(newGithubProfile);
+
+    return newGithubProfile;
+  }
+
+  findAll(): GithubProfile[] {
+    return [...this.githubProfiles];
+  }
+
+  findOne(id: number): GithubProfile {
+    const githubProfile = this.githubProfiles.find(
+      (profile) => profile.id === id,
+    );
+    if (!githubProfile) {
+      throw new HttpException('Githubprofile Not Found', HttpStatus.NOT_FOUND);
+    }
+    return { ...githubProfile };
+  }
+
+  //TODO Event Mapping
```

Comment:
Could the **todo** be written like this so that various tools pick up on them 

```ts
// @TODO: ...
```

Reply:
to be honest I just forgot to remove it.. :sweat_smile:

Reply:
oh ok lol 😂